batch/batch/server/server.py

```diff
@@ -876,7 +862,12 @@ def close_ttl(self):
     params = request.query
     user = userdata['ksa_name']
 
-    batches = [Batch.from_record(record) for record in await db.batch.get_record({'user': user})]
+    batches = []
+    for record in await db.batch.get_records_where({'user': user}):
+        batch = Batch.from_record(record)
+        if batch is not None:
```

Comment:
How can batch be None here?

Reply:
If the batch has been set to `deleted` in the database but the record still exists. I check for this in `Batch.from_record`

Reply:
You should push this into the database query.  Basically, no functions should return deleted rows now.